lib/updatehub/updatehub.c

```diff
@@ -185,33 +185,41 @@ static bool start_coap_client(void)
 		return false;
 	}
 
+	ret = -1;
+
 	ctx.sock = socket(addr->ai_family, SOCK_DGRAM, protocol);
 	if (ctx.sock < 0) {
 		LOG_ERR("Failed to create UDP socket");
-		return false;
+		goto error;
 	}
 
 #if defined(CONFIG_UPDATEHUB_DTLS)
 	if (setsockopt(ctx.sock, SOL_TLS, TLS_SEC_TAG_LIST,
 		       sec_list, sizeof(sec_list)) < 0) {
 		LOG_ERR("Failed to set TLS_TAG option");
-		return false;
+		goto error;
 	}
 
 	if (setsockopt(ctx.sock, SOL_TLS, TLS_PEER_VERIFY, &verify, sizeof(int)) < 0) {
 		LOG_ERR("Failed to set TLS_PEER_VERIFY option");
-		return false;
+		goto error;
 	}
 #endif
 
 	if (connect(ctx.sock, addr->ai_addr, addr->ai_addrlen) < 0) {
 		LOG_ERR("Cannot connect to UDP remote");
-		return false;
+		goto error;
 	}
 
 	prepare_fds();
 
-	return true;
+	ret = 0;
+
+error:
+	freeaddrinfo(addr);
+	addr = NULL;
```

Comment:
why are you setting `addr` to NULL ? This variable is allocated in the stack and not used anymore.

Reply:
Removed.